sdk/src/transport.rs

```diff
@@ -10,6 +15,12 @@ pub enum TransportError {
     TransactionError(#[from] TransactionError),
     #[error("transport custom error: {0}")]
     Custom(String),
+
+    #[error("transport send error: {0}")]
+    SendError(#[from] WriteError),
+
+    #[error("transport connect error: {0}")]
+    ConnectError(#[from] ConnectError),
```

Comment:
Again, we don't want quinn's details here

Reply:
I think the enums are now generic enough that they should apply to any quic implementation (connect error and send error), where basically if the implementation were switched, all we would need to do is replace the error type in the #from. I see something similar done for example at https://www.lpalmieri.com/posts/error-handling-rust/ and actually some of the other errors that were already in there previously do that, for example io::Error, which while admitedly probably more widely used is still in principle the same; if we switched to a new io lib for networking for example, we would need to swap that out. So the idea is that if we were to switch to another quic library we would basically switch for example from SendError(#[from] WriteError), to SendError(#[from] something_else), but the categories SendError and ConnectError should apply to basically any quic implementation